### PR TITLE
[WIP] DIP3 Masternode implementation

### PR DESCRIPTION
**Pull request by akshaynexus**
This is a WIP PR to add dip3 masternodes from dash,refactored to work with our codebase.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -68,7 +68,7 @@ AC_ARG_ENABLE([c++17],
   [AS_HELP_STRING([--enable-c++17],
   [enable compilation in c++17 mode (disabled by default)])],
   [use_cxx17=$enableval],
-  [use_cxx17=no])
+  [use_cxx17=yes])
```
Comment (TheQuantumPhysicist) [inline configure.ac:71]: This is relatively unsafe to do. C++17 has deprecated many things and removed many things. We have to check all the warnings that come up when compiling ghost with this flag.

File: configure.ac [hunk 2]
```diff
@@ -68,7 +68,7 @@ AC_ARG_ENABLE([c++17],
...
 
 dnl Require C++11 or C++17 compiler (no GNU extensions)
```
Comment (TheQuantumPhysicist) [inline configure.ac:73]: It looks like C++17 is not required here. Can you explain why you enabled it?

File: configure.ac (hunk 1 above)
Comment (akshaynexus) [reply configure.ac:71]: I did not see any new warnings or errors after doing so.

File: configure.ac (hunk 2 above)
Comment (akshaynexus) [reply configure.ac:73]: I enabled it to fix compile on a particular bls file,if you disable it you will get the error,i dont remember which file exactly it was.

**Pull request closed without merging**